replication-api/src/main/java/org/codice/ditto/replication/api/persistence/ReplicationPersistentStore.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ditto.replication.api.persistence;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+import org.codice.ditto.replication.api.data.Persistable;
+
+/** A ReplicationPersistentStore performs CRUD operations for replication. */
+public interface ReplicationPersistentStore {
+
+  /**
+   * Instantiates an implementation of the given interface which the client will then be responsible
+   * for populating with the proper fields.
+   *
+   * @param klass The interface to instantiate an implementation for
+   * @param <T> An interface that extends persistable
+   * @return An instance of an implementation of the specified interface
+   */
+  <T extends Persistable> T createPersistable(Class<T> klass);
```

Comment:
This basically comes back to whether we want setters in our interfaces right? I get the desire for small interfaces but having the setters on there makes it easy for us to update an object. If I want to edit a sites name I can just load it up, set the name, and save it, instead of creating a whole new site and saving that. If we do some refactoring this will also let us avoid having a dependency on the api and the impl componenents, we can just depend on the api.